### PR TITLE
🚮Restrict comments to 80 columns (exempt JSDoc annotations)

### LINKED ISSUE #15127: Consider linting max-len for comments
Comments over 80 columns don't trigger errors or warnings during `gulp lint`.  We could consider making that happen.

Links are already (separately) allowed via the `ignoreUrls` option specified [here](https://github.com/ampproject/amphtml/blob/master/.eslintrc#L70).

### PR DESCRIPTION
**Pull request by rsimha**
This PR does the following:
- Replaces `"ignoreComments": true` in `.eslintrc` with `"ignoreTrailingComments": true`
- Fixes all instances of long text comments using https://github.com/stkb/Rewrap
- Adds exemptions for long single-line JSDoc type annotations via `ignorePattern`
- Fixes a few incorrect type annotations ([here](https://travis-ci.org/ampproject/amphtml/jobs/378165206#L2090-L2099) and [here](https://travis-ci.org/ampproject/amphtml/jobs/378170756#L2090-L2159)) that were not being caught by `gulp check-types` because they were [split across lines](https://github.com/ampproject/amphtml/pull/14439/files#r188090503)

I've kicked off a separate effort to find and fix all JSDoc errors in our code: #15255

Fixes #15127 
To be followed by #15256

### CODE REVIEW

**Comment by rsimha** [comment]
@newmuis @jridgewell This is mostly done, except for a bunch of JSDoc type annotations that exceed 80 chars. Right now, I've added a bunch of patterns to `ignorePattern`, which doesn't work on multi-line type annotations that exceed 80 chars. I'm hoping there's a better way.

Got any ideas?

**Comment by rsimha** [comment]
@aghassemi @peterjosling I fixed a few [type annotation bugs](https://travis-ci.org/ampproject/amphtml/jobs/378170756#L2090-L2159) in `amp-next-page` that weren't being detected earlier. Can you review? See https://github.com/ampproject/amphtml/pull/15247/commits/dd25d19560a3547b4cf06f46b35033a63efba0e9

**Comment by rsimha** [comment]
Ping.

File: ads/google/a4a/google-data-reporter.js [hunk 1]
```diff
@@ -139,10 +139,10 @@ function setupPageLoadMetricsReporter_(ampElement) {
       'requests': {
         'fvt': 'https://csi.gstatic.com/csi?s=a4a' +
             `&c=${correlator}&met.a4a=` +
-	    'makeBodyVisible.${makeBodyVisible}~' +
```
Comment (jridgewell) [inline ads/google/a4a/google-data-reporter.js]: Why is this whitespace weird? It should be +4, right?

File: extensions/amp-next-page/0.1/next-page-service.js [hunk 2]
```diff
@@ -88,10 +88,7 @@ export class NextPageService {
     /** @private {?../../../src/service/viewport/viewport-impl.Viewport} */
     this.viewport_ = null;
 
-    /**
-     * @private
-     * {?../../../../src/service/position-observer/position-observer-impl.PositionObserver}
```
Comment (jridgewell) [inline extensions/amp-next-page/0.1/next-page-service.js]: Ooof.

File: extensions/amp-next-page/0.1/next-page-service.js [hunk 3]
```diff
@@ -276,7 +273,7 @@ export class NextPageService {
    * one.
    * @param {number} nextPage Index of the next unseen page to use as the first
    *     recommendation in the list.
-   * @return {Element} Container element for the recommendations.
+   * @return {!Element} Container element for the recommendations.
```
Comment (jridgewell) [inline extensions/amp-next-page/0.1/next-page-service.js]: Let's break this files changes into a new PR.

File: extensions/amp-next-page/0.1/next-page-service.js [hunk 4]
```diff
@@ -358,13 +355,12 @@ export class NextPageService {
...
    *     attached to.
-   * @param
-   * {!../../../src/service/position-observer/position-observer-worker.PositionInViewportEntryDef}
+   * @param {?../../../src/service/position-observer/position-observer-worker.PositionInViewportEntryDef}
    *     position Position of the current recommendation unit in the viewport.
    */
   positionUpdate_(i, position) {
     // We're only interested when the recommendations exit the viewport
-    if (position.positionRect !== null) {
+    if (!position || position.positionRect !== null) {
```
Comment (choumx) [inline extensions/amp-next-page/0.1/next-page-service.js]: Intended change or out of date base?

File: src/polyfills/domtokenlist-toggle.js [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
  *
  * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS-IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * distributed under the License is distributed on an "AS-IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
```
Comment (choumx) [inline src/polyfills/domtokenlist-toggle.js]: Should we be reformatting license text?

File: src/runtime.js [hunk 6]
```diff
@@ -430,7 +430,8 @@ export class MultidocManager {
    * @param {!Element} hostElement
    * @param {string} url
    * @param {!Object<string, string>|undefined} initParams
-   * @param {function(!Object, !ShadowRoot, !./service/ampdoc-impl.AmpDocShadow):!Promise} builder
+   * @param {function(!Object, !ShadowRoot,
+   * !./service/ampdoc-impl.AmpDocShadow):!Promise} builder
```
Comment (choumx) [inline src/runtime.js]: Revert this change?

File: src/base-element.js [hunk 7]
```diff
@@ -119,7 +119,7 @@ export class BaseElement {
     /** @public @const {!Element} */
     this.element = element;
     /*
-    \   \  /  \  /   / /   \     |   _  \     |  \ |  | |  | |  \ |  |  /  _____|
+    \   \  /  \  /   / /   \     |   _  \     |  \ |  | |  | |  \ |  |  /  ____|
```
Comment (newmuis) [inline src/base-element.js]: lol

File: ads/google/a4a/google-data-reporter.js (hunk 1 above)
Comment (rsimha) [reply ads/google/a4a/google-data-reporter.js]: Yeah, weird. Fixed so that the linter doesn't complain.

File: extensions/amp-next-page/0.1/next-page-service.js (hunk 4 above)
Comment (rsimha) [reply extensions/amp-next-page/0.1/next-page-service.js]: Intended. See https://github.com/ampproject/amphtml/pull/14439/files#r188090503

File: extensions/amp-next-page/0.1/next-page-service.js (hunk 3 above)
Comment (rsimha) [reply extensions/amp-next-page/0.1/next-page-service.js]: I've asked @aghassemi to review.

File: src/runtime.js (hunk 6 above)
Comment (rsimha) [reply src/runtime.js]: Intentional, since this breaks `gulp check-types` as written. See link above.

File: src/polyfills/domtokenlist-toggle.js (hunk 5 above)
Comment (rsimha) [reply src/polyfills/domtokenlist-toggle.js]: Good catch. Reverted.

File: extensions/amp-next-page/0.1/next-page-service.js (hunk 2 above)
Comment (rsimha) [reply extensions/amp-next-page/0.1/next-page-service.js]: Interestingly, this completely breaks `gulp check-types`, which sees just the `@private`, and ignores the type name. A more widespread fix is coming up.

File: extensions/amp-next-page/0.1/next-page-service.js (hunk 4 above)
Comment (aghassemi) [reply extensions/amp-next-page/0.1/next-page-service.js]: LGTM
